### PR TITLE
Android API: Kotlin callbacks alongside coroutines

### PR DESCRIPTION
**Pull request by jamesonwilliams**
**This update the Android API code snippets.**

While the Core Kotlin 0.1.1 library is incubating in beta, we will keep
the "vanilla" callback code snippets for Kotlin, too.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/graphqlapi/fragments/android/advanced-workflows/30_nested.md [hunk 1]
```diff
@@ -64,12 +64,23 @@ Then query for the Post:
...
+);
 ```
 
 </amplify-block>
-<amplify-block name="Kotlin">
+<amplify-block name="Kotlin - Callbacks">
+
+```kotlin
+Amplify.API.query(getPostWithCommentsRequest("[TODO_ID]"),
+    { Log.d("MyAmplifyApp", "Response = $response") },
```
Comment (raphkim) [inline docs/lib/graphqlapi/fragments/android/advanced-workflows/30_nested.md:77]: shouldn't it be `$it`?

File: docs/lib/graphqlapi/fragments/android/authz/30_multi.md [hunk 2]
```diff
@@ -13,7 +13,19 @@ Amplify.API.mutate(
...
 
 </amplify-block>
-<amplify-block name="Kotlin">
+<amplify-block name="Kotlin - Callbacks">
+
+```kotlin
+Amplify.API.mutate(
+    "[FRIENDLY-NAME-API-WITH-API-KEY]",
+    request,
+    { Log.i("MyAmplifyApp", "Mutation successful") },
```
Comment (raphkim) [inline docs/lib/graphqlapi/fragments/android/authz/30_multi.md:22]: `Log.i()` vs `Log.d()` has been used inconsistently throughout this doc.  Can we stick to one? (I prefer `Log.i()`)

File: docs/lib/restapi/fragments/android/fetch.md [hunk 3]
```diff
@@ -109,13 +126,29 @@ Amplify.API.get(options,
...
 
 </amplify-block>
-<amplify-block name="Kotlin">
+<amplify-block name="Kotlin - Callbacks">
 
 ```kotlin
 val request = RestOptions.builder()
     .addPath("/todo")
-    .addQueryParameters(mapOf("q", "test"))
+    .addQueryParameters(mapOf("q" to "test"))
```
Comment (raphkim) [inline docs/lib/restapi/fragments/android/fetch.md:134]: 🤯 whoa kotlin

**Review by raphkim — COMMENTED** [review]
Looks great besides a few comments!

File: docs/lib/restapi/fragments/android/fetch.md (hunk 3 above)
Comment (jamesonwilliams) [reply docs/lib/restapi/fragments/android/fetch.md:134]: haha yea!

File: docs/lib/graphqlapi/fragments/android/advanced-workflows/30_nested.md (hunk 1 above)
Comment (jamesonwilliams) [reply docs/lib/graphqlapi/fragments/android/advanced-workflows/30_nested.md:77]: YES! Thanks, will update.

Comment (jamesonwilliams) [inline docs/lib/graphqlapi/fragments/android/advanced-workflows/30_nested.md:77]:
```suggestion
    { Log.d("MyAmplifyApp", "Response = $it") },
```

**Review by raphkim — APPROVED** [review]

**Pull request merged**